File: tests/fixtures/supabase/supabase-storage-error-not-surfaced-broken/02-resume-upload.ts
import { supabase } from '../client.js';

export async function saveResume(userId: string, file: File, resumeUrl: string) {
  const { error: uploadError } = await supabase.storage.from('resumes').upload(`${userId}/resume.pdf`, file);
  if (!uploadError) {
    const { data } = supabase.storage.from('resumes').getPublicUrl(`${userId}/resume.pdf`);
    resumeUrl = data.publicUrl;
  }
  return supabase.from('profiles').update({ resume_url: resumeUrl }).eq('user_id', userId);
}

File: tests/fixtures/supabase/supabase-storage-error-not-surfaced-broken/02_resume_upload.py
from lib.client import supabase


def upload_resume(user_id: str, file):
    path = f"resumes/{user_id}.pdf"
    try:
        response = supabase.storage.from_("documents").upload(path=path, file=file)
    except Exception:
        ...
    return path
